native/src/main/java/io/ballerina/lib/avro/visitor/DeserializeVisitor.java

```diff
@@ -0,0 +1,411 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.avro.visitor;
+
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.types.ArrayType;
+import io.ballerina.runtime.api.types.Field;
+import io.ballerina.runtime.api.types.IntersectionType;
+import io.ballerina.runtime.api.types.MapType;
+import io.ballerina.runtime.api.types.RecordType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericEnumSymbol;
+import org.apache.avro.generic.GenericFixed;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.util.Utf8;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static io.ballerina.lib.avro.Utils.getMutableType;
+
+public class DeserializeVisitor implements IDeserializeVisitor {
+
+    public double visitDouble(Object data) {
+        if (data instanceof Float) {
+            return Double.parseDouble(data.toString());
+        }
+        return (double) data;
+    }
+
+    public BArray visitBytes(Object data) {
+        return ValueCreator.createArrayValue(((ByteBuffer) data).array());
+    }
+
+    public BArray visitFixed(Object data) {
+        GenericData.Fixed fixed = (GenericData.Fixed) data;
+        return ValueCreator.createArrayValue(fixed.bytes());
+    }
+
+    public BString visitString(Object data) {
+        return StringUtils.fromString(data.toString());
+    }
+
+    @SuppressWarnings({"unchecked", "deprecation"})
```

Comment:
Why are we suppressing `deprecated` here?

If we are using a deprecated API, we should consider moving to the alternative APIs, because the deprecated APIs might get removed in the future releases.